dcicutils/aws_utils.py

```diff
@@ -0,0 +1,261 @@
+""" Functions to support general aws operations
+
+    Function names should generally follow a convention to include a short string
+    to indicate the service that they are designed to operate with eg. 's3' for AWS s3.
+"""
+import boto3
+import logging
+import mimetypes
+
+from botocore.exceptions import ClientError
+# from .exceptions import ()
+from .s3_utils import s3Utils
+
+###########################
+# Config
+###########################
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+
+
+def s3_bucket_head(*, bucket_name, s3=None):
+    """ Gets head info for a bucket if it exists
+
+    :param bucket_name: name of the bucket - string
+    :param s3: AWS s3 client
+    :return: dict: head response or None
+    """
+    try:
+        s3 = s3 or s3Utils().s3
+        info = s3.head_bucket(Bucket=bucket_name)
+        return info
+    except ClientError:
+        return None
+
+
+def s3_bucket_exists(*, bucket_name, s3=None):
+    """ Does a bucket exist?
+
+    :param bucket_name: name of the bucket - string
+    :param s3: AWS s3 client
+    :return: boolean - True if exists, False if not
+    """
+    return bool(s3_bucket_head(bucket_name=bucket_name, s3=s3))
+
+
+def s3_bucket_object_count(bucket_name):
```

Comment:
You're using the `s3` convention inconsistently - I would recommend having it everywhere and writing a small helper that does the client initialization, if necessary.